js/script.js

```diff
@@ -1,59 +1,69 @@
-const bookCollection = document.getElementById('book-collection');
-const formInput = document.getElementById('book-form');
+class Library {  
+  constructor() {
+    this.bookData = [];
+    this.bookData = JSON.parse(localStorage.getItem('bookstoredata'));
+    this.formInput = document.getElementById('book-form');
+  }
+    
+  showSavedData () {
+    if (this.bookData) {
+      let bookList = '';
+      if (this.bookData.length > 0) {
+        bookList = '<div class="table-container">';        
+      }
+      else {
+        bookList = '<div>';
+      }
 
-let bookData = [];
+      bookList += '<table class="table table-hover table-striped"><tbody>';
+      this.bookData.forEach((item, i) => {
+        bookList += `        
+        <tr>
+          <td>"${item.title}" by ${item.author}</td>
+          <td><button type="submit" class="remove-button" id="${i}">Remove</button></td>
+        </tr>          
+        `;
+      });
+      bookList += '</tbody></table></div>';
+      const bookCollection = document.getElementById('book-collection');
+      bookCollection.innerHTML = bookList;   
+      const removeButtons = document.querySelectorAll('.remove-button');
+      removeButtons.forEach((item) => item.addEventListener('click', this.removeBook.bind(this)));    
+    }
+  }
 
-const removeBook = (event) => {
-  let id = event.target.id;
-  let modBookData = bookData.filter(function(element, i) {
-    return ((i != id) ? element : '');
-  });
-  bookData = modBookData;
-  storeData();
-  showSavedData();
-}
+  addBookData(event) {
+    event.preventDefault(); 
+    const title = document.getElementById('title');
+    const author = document.getElementById('author');
+    if (title.value.length > 0 && author.value.length > 0) {
+      this.bookData.push({
+        title: title.value,
+        author: author.value,
+      });
+      this.storeData();
+      this.showSavedData();
+      this.formInput.reset();
+    }    
+  }
```

Comment:
- [ ] There is a problem with this function, I noticed that users can't add a book. Kindly fix the issue shown in the image below.

![books](https://user-images.githubusercontent.com/53124548/213148061-60894bbc-2973-42ec-8605-d39229b61400.png)

Reply:
Hi @elisha2kyakpo1, thanks for the valuable feedback. But I'm having some doubts or trouble getting the error. When I try it in my system it's working fine and I can add and remove the books. So, can you please tell me how to reproduce this error?

Reply:
@lenkon, did you push everything? To be sure that the project works, I suggest you clone the repo and run it and let me know if it will work.

Reply:
Hi@elisha2kyakpo1, thanks for the prompt reply. Yes, I pushed everything to the repository. Now, I've done the following steps -
1. cloned the repo to an empty directory
2. changed the branch to "refactor-to-use-javascript-classes" from the main branch.
3. started the live server on port 5500
It's working fine and I can add, and remove the books. Also, it displays the stored books in the browser.

Could you please recheck and let me know how to reproduce the error if you still face the error?

Reply:
@lenkon if that's the case, I suggest you record a video (short) to show the next reviewer how the project works. In the video, you can show the current branch and the working app from the browser.

Reply:
@elisha2kyakpo1 alright, I would create a short demo video explaining the steps to perform the functions. And thanks for the feedback.

Reply:
Demo video showing the working of the project - https://www.loom.com/share/b894395a5f9b49caaf21018345d036dc

Reply:
👍